From 9ca3de0446fcd31b1561b333020ae0674a83bec7 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 9 Aug 2016 11:39:10 +0200
Subject: [PATCH] phpdbg_prompt: enclose phpdbg_load_module_or_extension() in
 HAVE_LIBDL

The phpdbg_load_module_or_extension() function relies on libdl
functionality, so it should be enclosed in HAVE_LIBDL to avoid the
following build failure when libdl support is not available:

php-7.0.9/sapi/phpdbg/phpdbg_prompt.c:(.text+0x1c20): undefined reference to `_DL_LOAD'
php-7.0.9/sapi/phpdbg/phpdbg_prompt.c:(.text+0x1c36): undefined reference to `_GET_DL_ERROR'
php-7.0.9/sapi/phpdbg/phpdbg_prompt.c:(.text+0x1c6a): undefined reference to `_DL_FETCH_SYMBOL'
php-7.0.9/sapi/phpdbg/phpdbg_prompt.c:(.text+0x1c7e): undefined reference to `_DL_FETCH_SYMBOL'
php-7.0.9/sapi/phpdbg/phpdbg_prompt.c:(.text+0x1dc6): undefined reference to `_DL_UNLOAD'

This function is only called in the "dl" debug command, inside an
HAVE_LIBDL condition.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 sapi/phpdbg/phpdbg_prompt.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sapi/phpdbg/phpdbg_prompt.c b/sapi/phpdbg/phpdbg_prompt.c
index 6f73d30..6768dc5 100644
--- a/sapi/phpdbg/phpdbg_prompt.c
+++ b/sapi/phpdbg/phpdbg_prompt.c
@@ -1019,6 +1019,7 @@ static int add_zendext_info(zend_extension *ext) /* {{{ */ {
 }
 /* }}} */
 
+#ifdef HAVE_LIBDL
 PHPDBG_API const char *phpdbg_load_module_or_extension(char **path, char **name) /* {{{ */ {
 	DL_HANDLE handle;
 	char *extension_dir;
@@ -1158,6 +1159,7 @@ quit:
 	DL_UNLOAD(handle);
 	return NULL;
 }
+#endif
 /* }}} */
 
 PHPDBG_COMMAND(dl) /* {{{ */
-- 
2.7.4

